lib/views/staging-view.js

```diff
@@ -391,20 +400,61 @@ export default class StagingView {
         await this.showMergeConflictFileForPath(selectedItem.filePath, {activate: true});
       }
     } else {
-      const pendingItem = this.getPendingFilePatchItem();
-      if (openNew || pendingItem) {
-        const activate = pendingItem === this.props.workspace.getActivePaneItem();
-        await this.showFilePatchItem(selectedItem.filePath, this.selection.getActiveListKey(), {activate});
+      if (openNew) {
+        // User explicitly asked to view diff, such as via click
+        await this.showFilePatchItem(selectedItem.filePath, this.selection.getActiveListKey(), {activate: false});
+      } else {
+        const panesWithStaleItemsToUpdate = this.getPanesWithStalePendingFilePatchItem();
+        if (panesWithStaleItemsToUpdate.length > 0) {
+          // Update stale items to reflect new selection
+          await Promise.all(panesWithStaleItemsToUpdate.map(async pane => {
+            await this.showFilePatchItem(selectedItem.filePath, this.selection.getActiveListKey(), {
+              activate: false,
+              pane,
+            });
+          }));
+        } else {
+          // Selection was changed via keyboard navigation, update pending item in active pane
+          const activePane = this.props.workspace.getCenter().getActivePane();
+          const activePendingItem = activePane.getPendingItem();
+          const activePaneHasPendingFilePatchItem = activePendingItem && activePendingItem.getRealItem &&
+            activePendingItem.getRealItem() instanceof FilePatchController;
```

Comment:
`instanceof` checks always make me feel vaguely uncomfortable. I think they raise my "OO code smell" instincts :smile:

(No useful alternatives to suggest here, though; we don't have much choice when we're dealing with arbitrary PaneItems.)